### PR TITLE
Resolution of Issue #202 and #139 and some random cleanup

### PR DESCRIPTION
**Pull request by ramensandwich**
I'm not 100% sure if the RBSM stuff is done correctly by this.  If someone from highlevel could take a look and let me know if I'm missing anything that would be great.

### CODE REVIEW

File: real_time/arduino_src/radio_buggy_mega/main.cpp [hunk 1]
```diff
@@ -65,6 +65,8 @@
 #define BRAKE_INDICATOR_PORT PORTE
 #define BRAKE_INDICATOR_PINN PE3 // arduino 5
 
+#define STATE_THRESHOLD 120 //This is a temporary change to eliminate magic number
```
Comment (TrevorDecker) [inline real_time/arduino_src/radio_buggy_mega/main.cpp]: ??? what is the more complete change?

Comment (ramensandwich) [inline real_time/arduino_src/radio_buggy_mega/main.cpp]: I was unsure if the two thresholds were related, so since they were the same value, I put in a temporary definition until someone complained that they were supposed to be treated differently

File: real_time/arduino_src/radio_buggy_mega/main.cpp [hunk 2]
```diff
@@ -139,13 +146,15 @@ void brake_raise() {
 
 // Drops the brake
 // Do not call before brake_init
-void brake_drop() {
+void brake_drop() 
+{
```
Comment (ihartwig) [inline real_time/arduino_src/radio_buggy_mega/main.cpp]: Can we not touch the lines that are not important to the function of this work?

File: real_time/arduino_src/radio_buggy_mega/main.cpp (hunk 1 above)
Comment (ihartwig) [inline real_time/arduino_src/radio_buggy_mega/main.cpp]: They are the same. It is a threshold used to make a pwm (servo) input binary. Can we name it PWM_STATE_THRESHOLD so it is grouped with the other PWM defines above?

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/mainFile.java [hunk 3]
```diff
@@ -61,4 +61,5 @@ public static void bringup_sim() throws Exception {
         String GpsKey = sm.newRealSensor(RealNodeEnum.GPS, config.COM_PORT_GPS_INTEGRATED, SensorChannel.GPS);
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/mainFile.java]: Codacy Issue found: [Avoid unused local variables such as 'GpsKey'.](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1262431887/issues/source?bid=2938665&fileBranchId=2938741#l61)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/RBSMNode.java [hunk 4]
```diff
@@ -97,28 +103,33 @@ public boolean matchDataSample(byte[] sample) {
...
-	public int peel(byte[] buffer, int start, int bytes_available) {
+	public int peel(byte[] buffer, int start, int bytes_available) 
+	{
 		// The Encoder sends 3 types of messages
 		//  - Encoder ticks since last message (keep)
 		//  - Number of ticks since last reset
 		//  - Timestamp since reset
 		RBPair rbp = RBSerial.peel(buffer, start, bytes_available);
-		switch(rbp.getNumberOfBytesRead()) {
+		switch(rbp.getNumberOfBytesRead())
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/RBSMNode.java]: Codacy Issue found: [A switch statement does not contain a break](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1262431913/issues/source?bid=2938665&fileBranchId=2938741#l126)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/mainFile.java [hunk 5]
```diff
@@ -61,4 +61,5 @@ public static void bringup_sim() throws Exception {
...
         String RBSMKey = sm.newRealSensor(RealNodeEnum.RBSM, config.COM_PORT_ENCODER, SensorChannel.ENCODER, SensorChannel.STEERING);
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/mainFile.java]: Codacy Issue found: [Avoid unused local variables such as 'RBSMKey'.](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1262431887/issues/source?bid=2938665&fileBranchId=2938741#l62)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/mainFile.java (hunk 3 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/mainFile.java]: Codacy Issue found: [Avoid unused local variables such as 'GpsKey'.](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1274095655/issues/source?bid=2938665&fileBranchId=2938741#l61)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/RBSMNode.java (hunk 4 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/RBSMNode.java]: Codacy Issue found: [A switch statement does not contain a break](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1274095682/issues/source?bid=2938665&fileBranchId=2938741#l126)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/mainFile.java (hunk 5 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/mainFile.java]: Codacy Issue found: [Avoid unused local variables such as 'RBSMKey'.](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1274095655/issues/source?bid=2938665&fileBranchId=2938741#l62)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/RBSMNode.java [hunk 6]
```diff
@@ -98,28 +104,33 @@ public boolean matchDataSample(byte[] sample) {
...
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/RBSMNode.java]: Codacy Issue found: [A switch statement does not contain a break](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1282713996/issues/source?bid=2938665&fileBranchId=2938741#l127)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/mainFile.java [hunk 7]
```diff
@@ -62,4 +62,5 @@ public static void bringup_sim() throws Exception {
...
         String LoggingKey = sm.newRealSensor(RealNodeEnum.LOGGING_BUTTON, "", SensorChannel.GUI_LOGGING_BUTTON);
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/mainFile.java]: Codacy Issue found: [Avoid unused local variables such as 'LoggingKey'.](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1282713969/issues/source?bid=2938665&fileBranchId=2938741#l63)

File: real_time/surface_src/java_src/settings.gradle [hunk 8]
```diff
@@ -1 +1 @@
-include 'BuggyRos', 'Alice', 'Tutorial'
+include 'BuggyRos', 'Alice'
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/settings.gradle]: why did you remove the tutorial?

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/mainFile.java [hunk 9]
```diff
@@ -62,4 +79,5 @@ public static void bringup_sim() throws Exception {
...
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/mainFile.java]: Codacy Issue found: [Avoid unused local variables such as 'LoggingKey'.](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1287212759/issues/source?bid=2938665&fileBranchId=2938741#l80)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/mainFile.java [hunk 10]
```diff
@@ -62,4 +79,5 @@ public static void bringup_sim() throws Exception {
         String RBSMKey = sm.newRealSensor(RealNodeEnum.RBSM, config.COM_PORT_ENCODER, SensorChannel.ENCODER, SensorChannel.STEERING);
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/mainFile.java]: Codacy Issue found: [Avoid unused local variables such as 'RBSMKey'.](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1287212759/issues/source?bid=2938665&fileBranchId=2938741#l79)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/mainFile.java [hunk 11]
```diff
@@ -13,40 +13,57 @@
...
+        else 
+        {
             Robot.getInstance();
         }   
     }
     
     //going to start by just connecting to the IMU
-    public static void bringup_sim() throws Exception {
-        if(config.logging){
+    public static void bringup_sim() throws Exception
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/mainFile.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1287212759/issues/source?bid=2938665&fileBranchId=2938741#l63)

File: real_time/surface_src/java_src/settings.gradle (hunk 8 above)
Comment (ramensandwich) [inline real_time/surface_src/java_src/settings.gradle]: I took it out and forgot to put it back :)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/mainFile.java (hunk 9 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/mainFile.java]: Codacy Issue found: [Avoid unused local variables such as 'LoggingKey'.](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1287826916/issues/source?bid=2938665&fileBranchId=2938741#l80)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/mainFile.java (hunk 10 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/mainFile.java]: Codacy Issue found: [Avoid unused local variables such as 'RBSMKey'.](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1287826916/issues/source?bid=2938665&fileBranchId=2938741#l79)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/mainFile.java (hunk 11 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/mainFile.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1287826916/issues/source?bid=2938665&fileBranchId=2938741#l63)

**Comment by TrevorDecker** [comment]
To what degree has this been tested? I am not completely up to speed on the low level code but I think this looks good to me. If it passes testing and @ihartwig  is ok with it I am good for a merge into master

**Comment by ramensandwich** [comment]
The startup information has been tested and seems to work on my home setup, so I imagine it should work just fine for the buggy as well.  The RBSM aspect, however, hasn't been tested.

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/mainFile.java (hunk 9 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/mainFile.java]: Codacy Issue found: [Avoid unused local variables such as 'LoggingKey'.](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1293101028/issues/source?bid=2938665&fileBranchId=2938741#l80)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/mainFile.java (hunk 10 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/mainFile.java]: Codacy Issue found: [Avoid unused local variables such as 'RBSMKey'.](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1293101028/issues/source?bid=2938665&fileBranchId=2938741#l79)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/mainFile.java (hunk 11 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/mainFile.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1293101028/issues/source?bid=2938665&fileBranchId=2938741#l63)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/RBSMNode.java [hunk 12]
```diff
@@ -163,18 +245,26 @@ public static JSONObject translatePeelMessageToJObject(String message) {
...
 		data.put("timestamp", messageData[1]);
 		
-		if(sensorName.equals("steering") || sensorName.equals("commanded_steering")) {
+		if(sensorName.equals("steering") || sensorName.equals("commanded_steering")) 
+		{
 			data.put("name", "Steering_" + sensorName);
 			params.put("angle", Float.valueOf(messageData[2]));
 		}
-		else if (sensorName.equals("encoder")) {
+		else if (sensorName.equals("fp_hash")) {
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/RBSMNode.java]: Codacy Issue found: [Position literals first in String comparisons](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1384444228/issues/source?bid=2938665&fileBranchId=2938741#l253)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/RBSMNode.java [hunk 13]
```diff
@@ -163,18 +245,26 @@ public static JSONObject translatePeelMessageToJObject(String message) {
...
+			data.put("name", "Fingerprint Hash");
+			params.put("hash_value", messageData[2]);
+			
+		}
+		else if (sensorName.equals("encoder"))
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/RBSMNode.java]: Codacy Issue found: [Position literals first in String comparisons](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1384444228/issues/source?bid=2938665&fileBranchId=2938741#l258)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/RBSMNode.java [hunk 14]
```diff
@@ -56,43 +67,87 @@
 	
 	Publisher statePub_enc;
 	Publisher statePub_pot;
+	
+	Subscriber messageSub_angle;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/RBSMNode.java]: Codacy Issue found: [Use explicit scoping instead of the default package private level](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1384444228/issues/source?bid=2938665&fileBranchId=2938741#l71)

**Comment by TrevorDecker** [comment]
This has been tested on the buggy and appears to work.

**Comment by ramensandwich** [comment]
As far as I can tell, the low level code does not introduce any new complexity.  Essentially it's a few new print statements (with various output redirects), and that shouldn't affect any of the time sensitive operations.  If there are no further objections or concerns, I'll go ahead and merge it
